optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java

```diff
@@ -0,0 +1,722 @@
+import static java.util.stream.Collectors.toMap;
+
+public class MeetingSchedulingXlsxFileIO extends AbstractXlsxSolutionFileIO<MeetingSchedule> {
+
+    // Meeting Parametrization:
+    protected static final String DO_ALL_MEETINGS_AS_SOON_AS_POSSIBLE = "Do all meetings as soon as possible";
+
+    protected static final String REQUIRED_AND_PREFERRED_ATTENDANCE_CONFLICT = "Required and preferred attendance conflict";
+    protected static final String PREFERRED_ATTENDANCE_CONFLICT = "Preferred attendance conflict";
+
+    protected static final String ROOM_CONFLICT = "Room conflict";
+    protected static final String DONT_GO_IN_OVERTIME = "Don't go in overtime";
+    protected static final String REQUIRED_ATTENDANCE_CONFLICT = "Required attendance conflict";
+    protected static final String REQUIRED_ROOM_CAPACITY = "Required room capacity";
+    protected static final String START_AND_END_ON_SAME_DAY = "Start and end on same day";
+
+    @Override
+    public MeetingSchedule read(File inputScheduleFile) {
+        try (InputStream in = new BufferedInputStream(new FileInputStream(inputScheduleFile))) {
+            XSSFWorkbook workbook = new XSSFWorkbook(in);
+            return new MeetingSchedulingXslxReader(workbook).read();
+        } catch (IOException | RuntimeException e) {
+            throw new IllegalStateException("Failed reading inputScheduleFile ("
+                                                + inputScheduleFile + ").", e);
+        }
+    }
+
+    private static class MeetingSchedulingXslxReader extends AbstractXslxReader<MeetingSchedule> {
+
+        public MeetingSchedulingXslxReader(XSSFWorkbook workbook) {
+            super(workbook);
+        }
+
+        public MeetingSchedule read() {
+            solution = new MeetingSchedule();
+            readConfiguration();
+            readPersonList();
+            readMeetingList();
+            readDayList();
+            readRoomList();
+
+            return solution;
+        }
+
+        private void readConfiguration() {
+            nextSheet("Configuration");
+            nextRow();
+            readHeaderCell("Meeting name");
+            nextRow(true);
+            readHeaderCell("Constraint");
+            readHeaderCell("Weight");
+            readHeaderCell("Description");
+
+            readIntConstraintLine(DO_ALL_MEETINGS_AS_SOON_AS_POSSIBLE, null, "");
+            readIntConstraintLine(REQUIRED_AND_PREFERRED_ATTENDANCE_CONFLICT, null, "");
+            readIntConstraintLine(PREFERRED_ATTENDANCE_CONFLICT, null, "");
+            readIntConstraintLine(ROOM_CONFLICT, null, "");
+            readIntConstraintLine(DONT_GO_IN_OVERTIME, null, "");
+            readIntConstraintLine(REQUIRED_ATTENDANCE_CONFLICT, null, "");
+            readIntConstraintLine(REQUIRED_ROOM_CAPACITY, null, "");
+            readIntConstraintLine(START_AND_END_ON_SAME_DAY, null, "");
+        }
+
+        private void readPersonList() {
```

Comment:
A suggestion - this method is quite long, breaking it into smaller methods (e.g. by reading individual cells) will make it far more readable.